### ISSUE TITLE
Develop techniques for specs over linked data structures

### ISSUE THREAD

**Issue by davidcok** [OpenJML/OpenJML#714]
See for example gitbug659

**Comment by saraswat** [comment]
Indeed, it would be very useful to see example singly/doubly-linked list code that works with OpenJML (with `-esc`). All the examples on this site are with arrays. The site points to Gary Leavens' site, and indeed there are a bunch of such classes there, e.g. [here](https://www.cs.ucf.edu/~leavens/JML-release/org/jmlspecs/samples/list/list1/). But they do not work out of the box (e.g. issues with `refine` and `import` tokens).  I can provide more details (am attempting to make them work) -- curious if anyone has already worked this out.
